test/boot_requirements_checker_test.rb

```diff
@@ -183,8 +184,65 @@ def find_vol(mount_point, volumes)
         context "with a partitions-based proposal" do
           let(:use_lvm) { false }
 
-          it "does not require any particular volume" do
-            expect(checker.needed_partitions).to be_empty
+          context "and MS-DOS partition table" do
+            let(:grub_partitions) { {} }
+
+            before do
+              allow(dev_sda).to receive(:partition_table).and_return(pt_msdos)
+            end
+
+            context "with sufficently large MBR gap" do
+              it "does not require any particular volume" do
+                expect(checker.needed_partitions).to be_empty
+              end
+            end
+
+            context "with too small MBR gap" do
+              before do
+                allow(analyzer).to receive(:mbr_gap).and_return(dev_sda.name => Yast::Storage::DiskSize.KiB(16))
+              end
+
+              it "raises an exception" do
+                expect { checker.needed_partitions }.to raise_error Yast::Storage::BootRequirementsChecker::Error
+              end
+            end
+
+            context "with no MBR gap" do
+              before do
+                allow(analyzer).to receive(:mbr_gap).and_return(dev_sda.name => Yast::Storage::DiskSize.KiB(0))
+              end
+
+              it "raises an exception" do
+                expect { checker.needed_partitions }.to raise_error Yast::Storage::BootRequirementsChecker::Error
+              end
+            end
+
+          end
+
+          context "and GPT partition table" do
+            before do
+              allow(dev_sda).to receive(:partition_table).and_return(pt_gpt)
+            end
+
+            context "if there is no GRUB partition" do
+              let(:grub_partitions) { {} }
+
+              it "requires a new GRUB partition" do
+                expect(checker.needed_partitions).to include(
+                  an_object_with_fields(partition_id: ::Storage::ID_GPT_BIOS, reuse: nil)
+                )
+              end
+            end
+
+            context "if there is already a GRUB partition" do
+              let(:grub_partitions) { { dev_sda.name => [analyzer_part(dev_sda.name + "2")] } }
+
+              it "does not propose a new GRUB partition" do
```

Comment:
I was thinking that we probably should also do this reuse stuff here and then we'd have to look for a partition with reuse flag. But the checker code atm doesn't do this and I didn't want to change it in this pr, so I coded it this way.